src/Uno.Foundation.Runtime.WebAssembly/Helpers/DisposableAction.cs

```diff
@@ -0,0 +1,42 @@
+// ******************************************************************
+// Copyright � 2015-2018 nventive inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ******************************************************************
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Uno.Foundation.Runtime.WebAssembly.Helpers
+{
+    internal class DisposableAction : IDisposable
+    {
+        public DisposableAction(Action action)
+        {
+            Action = action;
+        }
+
+        public Action Action { get; private set; }
```

Comment:
nit: Setter not actually needed
```suggestion
        public Action Action { get; }
```